mlflow/genai/optimize/base.py

```diff
@@ -0,0 +1,158 @@
+import inspect
+from typing import TYPE_CHECKING, Optional
+
+from mlflow.entities.model_registry import Prompt
+from mlflow.exceptions import MlflowException
+from mlflow.genai.evaluation.utils import (
+    _convert_to_legacy_eval_set,
+)
+from mlflow.genai.optimize.optimizer import _BaseOptimizer, _DSPyMIPROv2Optimizer
+from mlflow.genai.optimize.types import (
+    OBJECTIVE_FN,
+    LLMParam,
+    OptimizerConfig,
+    PromptOptimizationResult,
+)
+from mlflow.genai.scorers import Scorer
+from mlflow.tracking._model_registry.fluent import load_prompt, register_prompt
+from mlflow.utils.annotations import experimental
+
+if TYPE_CHECKING:
+    from genai.evaluation.utils import EvaluationDatasetTypes
+
+_ALGORITHMS = {"DSPy/MIPROv2": _DSPyMIPROv2Optimizer}
+
+
+@experimental
+def optimize_prompt(
+    *,
+    agent_llm: LLMParam,
+    prompt: str,
+    train_data: "EvaluationDatasetTypes",
+    scorers: list[Scorer],
+    objective: Optional[OBJECTIVE_FN] = None,
+    eval_data: Optional["EvaluationDatasetTypes"] = None,
+    optimizer_config: Optional[OptimizerConfig] = None,
+) -> PromptOptimizationResult:
+    """Optimize a LLM prompt using the given dataset and evaluation metrics.
+    Currently, only supports MIPROv2 optimizer of DSPy.
+
+    Args:
+        agent_llm: Parameters for the agent LLM. The model name must be specified in the format
+            `<provider>/<model>`.
+        prompt: The URI of the MLflow prompt to optimize. This will be used as the initial
```

Comment:
```suggestion
        prompt_uri: The URI of the MLflow prompt to optimize. This will be used as the initial
```
nit: `prompt: str` sounds like it accepts the prompt text itself